# Baseline: OpenVela NuttX 76354c637858ecb0aa4601629327acb6f44a26bb
#
# Outgoing ACL buffers have one byte less head reserve than bt_receive().
# Do not negotiate the extra byte: incoming ATT data must fit after H4, ACL
# and L2CAP headers as represented by BLUETOOTH_MAX_MTU.
--- a/wireless/bluetooth/bt_att.c
+++ b/wireless/bluetooth/bt_att.c
@@ -498,7 +498,14 @@ static uint8_t att_mtu_req(FAR struct bt_conn_s *conn,
    * extra byte for ATT header.
    */
 
   maxmtu = bt_buf_tailroom(buf) + 1;
+  if (maxmtu > BLUETOOTH_MAX_MTU)
+    {
+      /* bt_receive() reserves the H4 header; incoming ATT must fit too. */
+
+      maxmtu = BLUETOOTH_MAX_MTU;
+    }
+
   if (mtu > maxmtu)
     {
       mtu = maxmtu;
